internal/cmd/controller/gitops/reconciler/gitjob_test.go

```diff
@@ -2426,61 +2423,21 @@ func TestDrivenScanSeparator(t *testing.T) {
 					Bundles: test.bundles,
 				},
 			}
-			separator := getDrivenScanSeparator(*gitrepo)
-			if !test.expectedRandomResult {
+			separator, err := getDrivenScanSeparator(*gitrepo)
+			if !test.expectError {
+				if err != nil {
+					t.Errorf("unexpected error: %v", err)
+				}
 				if test.expectedResult != separator {
 					t.Errorf("expecting separator to be: %q, got: %q", test.expectedResult, separator)
 				}
 			} else {
-				// Verify that the string has the desired length and
-				// is composed only of the specified characters.
-				if len(separator) != test.expectedResultRandomLength {
-					t.Errorf("expecting separator length to be: %d, got: %d", test.expectedResultRandomLength, len(separator))
-				}
-				for _, char := range separator {
-					if !strings.ContainsRune(bundleOptionsSeparatorChars, char) {
-						t.Errorf("expecting separator to only contain characters from the set: %q, got: %q", bundleOptionsSeparatorChars, separator)
-					}
-				}
-				for _, b := range gitrepo.Spec.Bundles {
-					// use the separator and verify we get the expected items
-					concatStr := b.Base + separator + b.Options
-					items := strings.Split(concatStr, separator)
-					if len(items) != 2 {
-						t.Errorf("expecting to have exactly 2 items, got: %d", len(items))
-					}
-					if items[0] != b.Base {
-						t.Errorf("expecting first item to be: %q, got: %q", b.Base, items[0])
-					}
-					if items[1] != b.Options {
-						t.Errorf("expecting second item to be: %q, got: %q", b.Options, items[1])
-					}
-				}
-				// Take it a step further and use the random separator in the bundle definitions.
-				// This will verify that the function returns a longer separator.
-				gitrepo = &fleetv1.GitRepo{
-					Spec: fleetv1.GitRepoSpec{
-						Bundles: []fleetv1.BundlePath{
-							{
-								Base:    fmt.Sprintf("test?one:%stwo", separator),
-								Options: "options.yaml",
-							},
-							{
-								Base:    "test,one<>",
-								Options: "options2|.yaml",
-							},
-						},
-					},
+				expectedErrorMessage := fmt.Sprintf("bundle base and/or options paths contain all possible characters from %q, please update those paths to remedy this", bundleOptionsSeparatorChars)
```

Comment:
nit: I'm not sure we _need_ to check against the full error message, as that could make our tests brittle. Checking that an error occurs could be enough.

Reply:
Oh, that was because as we're giving extra info to the user I thought it was good to check for the full error we return instead to just check that an error happened.